pkg/operator/ceph/csi/secrets.go

```diff
@@ -203,5 +259,51 @@ func CreateCSISecrets(context *clusterd.Context, clusterInfo *client.ClusterInfo
 		return errors.Wrap(err, "failed to create kubernetes csi secret")
 	}
 
+	err = updateCephStatusWithCephxStatus(context, clusterInfo, clusterSpec, clusterNamespaced, shouldRotateCephxKeys)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func updateCephStatusWithCephxStatus(context *clusterd.Context, clusterInfo *client.ClusterInfo, clusterSpec *cephv1.ClusterSpec, name types.NamespacedName, shouldRotateCephxKeys bool) error {
+	cephCluster := &cephv1.CephCluster{}
+	err := retry.OnError(
+		retry.DefaultRetry,
+		func(err error) bool {
+			return !kerrors.IsNotFound(err)
+		},
+		func() error {
+			if getErr := context.Client.Get(clusterInfo.Context, name, cephCluster); getErr != nil {
+				if kerrors.IsNotFound(getErr) {
+					logger.Errorf("CephCluster %q resource not found in namespace %q. Ignoring since object must be deleted. %v", name, name.Namespace, getErr)
+					return getErr
+				}
+				return errors.Wrapf(getErr, "Failed to retrieve cephCluster %q in namespace %q to update csi cephx status.", name, name.Namespace)
+			}
+			return nil
+		})
+	if err != nil {
+		return errors.Wrapf(err, "After multiple retries failed to get Ceph cluster %q status in namespace %q", name.Name, name.Namespace)
+	}
+
+	cephxStatus := keyring.UpdatedCephxStatus(shouldRotateCephxKeys, clusterSpec.Security.CephX.Daemon, clusterInfo.CephVersion, cephCluster.Status.Cephx.CSI)
+	cephCluster.Status.Cephx.CSI = cephxStatus
+
+	err = retry.RetryOnConflict(
+		retry.DefaultRetry,
+		func() error {
+			if getErr := reporting.UpdateStatus(context.Client, cephCluster); getErr != nil {
+				logger.Errorf("Failed to update cephCluster %q to update csi cephx status to %q in namespace %q with error %v", name.Name, cephxStatus, name.Namespace, getErr)
+				return getErr
+			}
+			return nil
+		})
+	if err != nil {
+		return errors.Wrapf(err, "After multiple retries failed to update Ceph cluster %q status with CSI Cephx status in namespace %q", name.Name, name.Namespace)
+	}
```

Comment:
```suggestion
	err = retry.RetryOnConflict(
		retry.DefaultRetry,
		func() error {
			if getErr := reporting.UpdateStatus(context.Client, cephCluster); getErr != nil {
				logger.Errorf("Failed to update cephCluster %q to update csi cephx status to %q in namespace %q with error %v", name.Name, cephxStatus, name.Namespace, getErr)
				return getErr
			}
			return nil
		})
	if err != nil {
		return errors.Wrapf(err, "After multiple retries failed to update Ceph cluster %q status with CSI Cephx status in namespace %q", name.Name, name.Namespace)
	}
```

@BlaineEXE same here, let me know if `retry.OnConflicts` is good here or I can use `OnError` with retry on all error

Reply:
`retry.OnError` seems like the most safe approach. But there should be only a single retry loop, not two. More here: https://github.com/rook/rook/pull/16040#discussion_r2167260708